File: Data Structures & Algorithms/binary-tree-right-side-view/submission-2.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def rightSideView(self, root: Optional[TreeNode]) -> List[int]:
        from collections import deque

        if not root:
            return []

        q = deque()
        q.append((root, 0))

        prev = (None, -1)
        res = []
        while q:

            node, level = q.popleft()
            if level > prev[1] and prev[1] != -1:
                res.append(prev[0].val)
            prev = (node, level)
            if node.left is not None:
                q.append((node.left, level + 1))
            if node.right is not None:
                q.append((node.right, level + 1))

        res.append(prev[0].val)
        return res

File: Data Structures & Algorithms/kth-smallest-integer-in-bst/submission-2.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */

class Solution {
private:
    vector<int> bstList (TreeNode * root){
        vector <int> res;
        if (root == nullptr){
            return res;
        }

        vector<int> left = bstList(root -> left);
        vector<int> right = bstList(root -> right);

        for (const int each_n : left){
            res.push_back(each_n);
        }
        
        res.push_back(root->val);

        for (const int each_n : right){
            res.push_back(each_n);
        }

        return  res;
    }
public:
    int kthSmallest(TreeNode* root, int k) {


        
        vector <int> lstBst = bstList(root);

        return lstBst[k-1];
    }
};
